core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,360 @@
+
+    @Test
+    def testConsumerNeedsAuthorization() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      //TODO: Ideally we would want to test that when consumerGroup permission is not present we still get an AuthorizationException
+      //but the consumer fetcher currently waits forever for the consumer metadata to become available.
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+      sendRecords(1, tp)
+      try {
+        this.consumers.head.assign(List(tp).asJava)
+        consumeRecords(this.consumers.head)
+        Assert.fail("should have thrown exception")
+      } catch {
+        case e: ApiException => Assert.assertEquals("Not authorized to read from topic-0", e.getMessage)
+      }
+    }
+
+    @Test
+    def testAllowingReadOnTopicAndGroupAllowsReading() {
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), topicResource)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), topicResource)
+      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), groupResource)
+      sendRecords(1, tp)
+      this.consumers.head.assign(List(tp).asJava)
+      consumeRecords(this.consumers.head)
+    }
+  
+//    TODO: The following test goes into an infinite loop as consumer waits for consumer metadata to be propogated for ever.
+//    @Test
+//    def testCreatePermissionNeededToReadFromNonExistentTopic() {
+//      val newTopic = "newTopic"
+//      val topicPartition = new TopicPartition(newTopic, 0)
+//      val newTopicResource = new Resource(Topic, newTopic)
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), newTopicResource)
+//      try {
+//        this.consumers(0).assign(List(topicPartition).asJava)
+//        consumeRecords(this.consumers(0))
+//        Assert.fail("should have thrown exception")
+//      } catch {
+//        case e: TimeoutException =>
+//        //TODO need to update the consumer so if no metadata is propagated it throws server side exception instead of trying forever.
+//        case e: Exception => Assert.fail("Only timeout exception should be thrown.")
+//      }
+//
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), newTopicResource)
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+//      try {
+//        sendRecords(numRecords, topicPartition)
+//        consumeRecords(this.consumers(0))
+//      } catch {
+//        case e: Exception => Assert.fail("No exception should be thrown.")
+//      }
+//    }
+
+  def removeAllAcls() = {
+    authorizer.getAcls().keys.foreach { resource =>
+      authorizer.removeAcls(resource)
+      TestUtils.waitAndVerifyAcls(Set.empty[Acl], authorizer, resource)
```

Comment:
To really makes sure that the acl is propagated to the broker, it seems that we need to verify against the authorizer instance in the broker.